database/migrations/0001_init_podcasting_20_database.sql

```diff
@@ -0,0 +1,844 @@
+
+--** CHANNEL > ABOUT
+
+-- various <channel> child data from multiple tags
+CREATE TABLE channel_about (
+    id SERIAL PRIMARY KEY,
+    channel_id INTEGER NOT NULL REFERENCES channel(id) ON DELETE CASCADE,
+    author varchar_normal, -- <itunes:author> and <author>
+    episode_count INTEGER, -- aggregated count for convenience
+    explicit BOOLEAN, -- <itunes:explicit>
+    itunes_type_id INTEGER REFERENCES channel_itunes_type(id),
+    language varchar_short NOT NULL, -- <language>
+    website_link_url varchar_url -- <link>
+);
+
+--** CHANNEL > CATEGORY
+
+CREATE TABLE channel_category (
+    id SERIAL PRIMARY KEY,
+    channel_id INTEGER NOT NULL REFERENCES channel(id) ON DELETE CASCADE,
+    parent_id INTEGER REFERENCES channel_category(id) ON DELETE CASCADE
+);
+
+--** CHANNEL > CHAT
+
+-- <channel> -> <podcast:chat>
+CREATE TABLE channel_chat (
+    id SERIAL PRIMARY KEY,
+    channel_id INTEGER NOT NULL REFERENCES channel(id) ON DELETE CASCADE,
+    server varchar_fqdn NOT NULL,
+    protocol varchar_short,
+    account_id varchar_normal,
+    space varchar_normal
+);
+
+--** CHANNEL > DESCRIPTION
+
+-- <channel> -> <description> AND possibly other tags that contain a description
+CREATE TABLE channel_description (
+    id SERIAL PRIMARY KEY,
+    channel_id INTEGER NOT NULL REFERENCES channel(id) ON DELETE CASCADE,
+    UNIQUE (channel_id),
+    value varchar_long NOT NULL
+);
+
+--** CHANNEL > FUNDING
+
+-- <channel> -> <podcast:funding>
+CREATE TABLE channel_funding (
+    id SERIAL PRIMARY KEY,
+    channel_id INTEGER NOT NULL REFERENCES channel(id) ON DELETE CASCADE,
+    url varchar_url NOT NULL,
+    title varchar_normal
+);
+
+--** CHANNEL > IMAGE
+
+-- <channel> -> <podcast:image> AND all other image tags in the rss feed
+CREATE TABLE channel_image (
+    id SERIAL PRIMARY KEY,
+    channel_id INTEGER NOT NULL REFERENCES channel(id) ON DELETE CASCADE,
+    url varchar_url NOT NULL,
+    image_width_size INTEGER, -- <podcast:image> must have a width specified, but older image tags will not, so allow null.
+
+    -- If true, then the image is hosted by us in a service like S3.
+    -- When is_resized images are deleted, the corresponding image in S3
+    -- should also be deleted.
+    is_resized BOOLEAN DEFAULT FALSE
+);
+
+--** CHANNEL > INTERNAL SETTINGS
+
+CREATE TABLE channel_internal_settings (
+    id SERIAL PRIMARY KEY,
+    channel_id INTEGER NOT NULL REFERENCES channel(id) ON DELETE CASCADE,
+    -- needed to approve which web domains can override the player with query params.
+    -- this prevents malicious parties from misrepresenting the podcast contents on another website.
+    embed_approved_media_url_paths TEXT
+);
+
+--** CHANNEL > LICENSE
+
+-- <channel> -> <podcast:license>
+CREATE TABLE channel_license (
+    id SERIAL PRIMARY KEY,
+    channel_id INTEGER NOT NULL REFERENCES channel(id) ON DELETE CASCADE,
+    UNIQUE (channel_id),
+    type varchar_normal NOT NULL,
+    url varchar_url NOT NULL
```

Comment:
In a few of the columns you are creating for urls, you are using both TEXT and varchar. Both effectively will do the same thing-TEXT just has unlimited characters. I would stick with one and be consistent.

Reply:
I just checked in `0001_` and I see 7 `TEXT` results, and all of them are used either 1) for columns that function like enums, where a finite number of values are possible, so the unlimited nature of `TEXT` shouldn't pose a problem, or 2) `TEXT` is used in:

```

CREATE TABLE channel_internal_settings (
    id SERIAL PRIMARY KEY,
    channel_id INTEGER NOT NULL REFERENCES channel(id) ON DELETE CASCADE,
    -- needed to approve which web domains can override the player with query params.
    -- this prevents malicious parties from misrepresenting the podcast contents on another website.
    embed_approved_media_url_paths TEXT
);
```

Which is a field only admins can set internally, so I wasn't worried about it filling up with data.

That said, in 1) these tables are kind of like enums? I glanced online and see `ENUM` is a type in Postgres. Should I be leveraging enums somehow with these 1) tables?